tools/buildsys/src/main.rs

```diff
@@ -270,6 +274,86 @@ fn build_variant() -> Result<()> {
     Ok(())
 }
 
+// Get all repo paths that make up the source directories for a given variant.
+// Expects BUIDLSYS_ROOT_DIR and BUILDSYS_VARIANT environment variables to be
+// set to know the path to the root of the repo and the variant to inspect.
+fn variant_paths() -> Result<()> {
+    let root_dir: PathBuf = getenv("BUILDSYS_ROOT_DIR")?.into();
+    let root_dir = root_dir.canonicalize().unwrap_or(root_dir);
+
+    let variant = getenv("BUILDSYS_VARIANT")?;
+    let manifest_dir = root_dir.join("variants").join(variant);
+
+    let manifest =
+        ManifestInfo::new(manifest_dir.join(MANIFEST_FILE)).context(error::ManifestParseSnafu)?;
+
+    let mut results: HashSet<PathBuf> = HashSet::new();
+
+    // manifest.included_packages gives the RPM package names, which doesn't always match with the local
+    // package directory name. We need to use the [*dependencies] sections to find the real packages.
+    results.insert(manifest_dir.clone());
+    let dependencies = manifest.local_dependency_paths(&manifest_dir);
+    for dependency in dependencies {
+        if add_package_paths(&root_dir, &dependency, &mut results).is_err() {
+            eprintln!("Error reading package information from '{}'", &dependency);
+        }
+    }
+
+    // Print out our collected paths
+    for path in results {
+        // Normalize the paths to be relative to the repo root
+        let clean_path = path.strip_prefix(&root_dir).unwrap_or(&path);
+        println!("{}", clean_path.display());
+    }
+
+    // Add the hardcoded files and paths that we always want
+    println!("Makefile.toml");
+    println!("Twoliter.toml");
+    println!(".github/actions");
+    println!(".github/workflows");
+
+    Ok(())
+}
+
+// Given a directory path the contains a Cargo.toml file, this inspects the package metadata to determine
+// what its dependencies are and adds any relevant paths to the `paths` collection.
+fn add_package_paths(
+    root_dir: &PathBuf,
+    package_dir: &str,
+    paths: &mut HashSet<PathBuf>,
+) -> Result<()> {
+    let package_dir = root_dir.join(package_dir);
+    paths.insert(package_dir.clone());
+
+    let manifest_path = root_dir.join(&package_dir).join(MANIFEST_FILE);
+    if !manifest_path.exists() {
+        // This is normal if it is a Go package or something similar
+        return Ok(());
+    }
+
+    let manifest = ManifestInfo::new(manifest_path).context(error::ManifestParseSnafu)?;
+
+    // Add our package dependencies
+    let dependencies = manifest.local_dependency_paths(&package_dir);
+    for dependency in dependencies {
+        if add_package_paths(root_dir, &dependency, paths).is_err() {
+            eprintln!("Error reading package information from '{}'", &dependency);
+        }
+    }
+
+    // Make sure we get any references local sources
+    if let Some(sources) = manifest.source_groups().cloned() {
+        for source in sources {
+            let source_path = format!("sources/{}", &source);
```

Comment:
To keep `source` as a `PathBuf`:
```suggestion
            let source_path = format!("sources/{}", &source.display());
```